### PR TITLE
chore: fix `tsd` and e2eTests

### PR DESCRIPTION
**Pull request by yasinmiran**

### CODE REVIEW

File: services/cega-mock/main.go [hunk 1]
```diff
@@ -75,6 +76,7 @@
 		http.Error(w, "Unauthorized", http.StatusUnauthorized)
 		return
 	}
+	fmt.Println("New authentication request", authHeader)
```
[the remark this answers is not in the document]
Comment (yasinmiran) [reply services/cega-mock/main.go]: fixed in [a77c5a4](https://github.com/ELIXIR-NO/FEGA-Norway/pull/189/commits/a77c5a4b74fe85cd7c0f1efc0638b7c7acfee19e)

File: services/cega-mock/main.go [hunk 2]
```diff
@@ -88,8 +90,10 @@
 		return
 	}
 	instance, passwd := parts[0], parts[1]
+	fmt.Println("Decoded auth parts", instance, passwd)
```
[the remark this answers is not in the document]
Comment (yasinmiran) [reply services/cega-mock/main.go]: fixed in [a77c5a4](https://github.com/ELIXIR-NO/FEGA-Norway/pull/189/commits/a77c5a4b74fe85cd7c0f1efc0638b7c7acfee19e)

File: lib/tsd-file-api-client/src/main/java/no/uio/ifi/tc/TSDFileAPIClient.java [hunk 3]
```diff
@@ -597,7 +600,7 @@ public TSDFileAPIClient build() {
 
       TSDFileAPIClient tsdFileAPIClient = new TSDFileAPIClient(httpClient);
 
-      tsdFileAPIClient.protocol = this.secure == null ? "https" : (this.secure ? "https" : "http");
+      tsdFileAPIClient.protocol = "http";
```
Comment (kjellp) [inline lib/tsd-file-api-client/src/main/java/no/uio/ifi/tc/TSDFileAPIClient.java:603]: Will this not break https operation of the TSD File API client when running with SSL in operation/staging ?

File: lib/tsd-file-api-client/src/main/java/no/uio/ifi/tc/model/pojo/ResumableUpload.java [hunk 4]
```diff
@@ -32,6 +32,6 @@ public class ResumableUpload extends TSDFileAPIResponse {
   @SerializedName("max_chunk")
   private BigInteger maxChunk;
 
-  @SerializedName("md5sum")
+  @SerializedName("md5Sum")
```
Comment (kjellp) [inline lib/tsd-file-api-client/src/main/java/no/uio/ifi/tc/model/pojo/ResumableUpload.java:35]: This now works with the TSD stub code, but are we sure it's compatible with the real API? Could approving it here now make it break with the "real world"?

File: services/tsd-api-mock/src/main/java/no/elixir/tsdapimock/resumables/ResumableUploadDto.java [hunk 5]
```diff
@@ -11,4 +11,4 @@ public record ResumableUploadDto(
     @JsonProperty("previous_offset") BigInteger previousOffset,
     @JsonProperty("next_offset") BigInteger nextOffset,
     @JsonProperty("max_chunk") BigInteger maxChunk,
-    @JsonProperty("md5sum") String md5Sum) {}
+    @JsonProperty("md5Sum") String md5Sum) {}
```
Comment (kjellp) [inline services/tsd-api-mock/src/main/java/no/elixir/tsdapimock/resumables/ResumableUploadDto.java:14]: As commented above, are we sure we mimic the correct behaviour in the realworld TSD API here?

**Review by kjellp — COMMENTED** [review]
I've made a few comments where suggested changes affect how client services calls or processes responses from a remote API, where the remote API is a mock service in our e2etests and will have a real world counter part when deployed in Staging/Production environment. These changes has the potential to break the interaction with the real-world counterpart services.

File: lib/tsd-file-api-client/src/main/java/no/uio/ifi/tc/TSDFileAPIClient.java (hunk 3 above)
Comment (yasinmiran) [reply lib/tsd-file-api-client/src/main/java/no/uio/ifi/tc/TSDFileAPIClient.java]: Good catch! thank you, Kjell.

I have resolved this with commit [d7452e9](https://github.com/ELIXIR-NO/FEGA-Norway/pull/189/commits/d7452e937993853b994279fbe7d8b60e960f82a1)

Added the missing `secure()` builder method and added the ternary condition. the default value is now `true` if the value is not set. we override this in the e2eTests setup by passing an ENV variable.

**Review by kjellp — APPROVED** [review]
I tested the latest fixes/changes by building all the services locally and export the tsd-proxy image (with the updated tsd-file-api-client included) to the tryggve.tsd.usit.uio.no server to test lega-commander listing and uploads in real life. And nothing broke :)

File: lib/tsd-file-api-client/src/main/java/no/uio/ifi/tc/model/pojo/ResumableUpload.java (hunk 4 above)
Comment (yasinmiran) [reply lib/tsd-file-api-client/src/main/java/no/uio/ifi/tc/model/pojo/ResumableUpload.java]: Temporary fix with [eb97539](https://github.com/ELIXIR-NO/FEGA-Norway/pull/189/commits/eb97539c643fa3d77913c28ac5145d7ca1172418)

File: services/tsd-api-mock/src/main/java/no/elixir/tsdapimock/resumables/ResumableUploadDto.java (hunk 5 above)
Comment (yasinmiran) [reply services/tsd-api-mock/src/main/java/no/elixir/tsdapimock/resumables/ResumableUploadDto.java]: Temporary fix with [eb97539](https://github.com/ELIXIR-NO/FEGA-Norway/pull/189/commits/eb97539c643fa3d77913c28ac5145d7ca1172418)

**Pull request merged**